### ISSUE TITLE
Adjust generated documentation for deprecated symbols

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#17948]
It would be handy for deprecated symbols to automatically indicate that they are deprecated in chpldoc's output.  I was thinking something along the lines of adding a `.. warning::` with the provided error message.

Users might want to explicitly link to the new symbol to use if there is one.  It might not be too hard to detect if there is already a `.. warning` with deprecation mentioned in some form in the documentation and suppress the generated one if so (I think it would be much more difficult to adjust the generated message to detect if there is an alternative symbol listed and link to that, so wouldn't expect to pursue that path)

**Comment by mppf** [comment]
Another thing that e.g. Python docs have and I use regularly is information about when the API function was added / changed. It is probably reasonable to put that off until we claim some stability but if that's not too far in the future maybe we can consider what it would take to have chpldoc include such information. (Maybe it is just to be added in the docs comments themselves but that seems unlikely to be a practice universally followed without some kind of checking).

**Comment by lydia-duncan** [comment]
I agree that something like that would be useful but I think it's worth spawning its own issue for :)
